server/openapi_server/controllers/teacher_controller.py

```
@@ -0,0 +1,79 @@
+
+teachers_router = APIRouter()
+
+@teachers_router.get("/", response_model=list[Teacher])
+def get_all_teachers():
+    """Retrieve all teachers."""
+    try:
+        collection = mongo_db.get_collection("teachers")
+        teachers = list(collection.find({}, {'_id': 0}))  # Exclude `_id` field from the results
+        return [Teacher(**teacher) for teacher in teachers]
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)